Framework/PythonInterface/plugins/algorithms/LoadWANDSCD.py

```diff
@@ -179,7 +302,41 @@ def PyExec(self):
         setGoniometer_alg.setProperty("Average", False)
         setGoniometer_alg.execute()
 
-        self.setProperty("OutputWorkspace", outWS)
+        return outWS
+
+    def normalize(
+        self,
+        data: IMDHistoWorkspace,
+        norm: IMDHistoWorkspace,
+        normalize_by: str,
+    ) -> IMDHistoWorkspace:
+        """
+        Normalize the given data with given Va normalization workspace
+        """
+        # prep
+        norm_replicated = ReplicateMD(ShapeWorkspace=data, DataWorkspace=norm)
+        data = DivideMD(LHSWorkspace=data, RHSWorkspace=norm_replicated)
+        # find the scale
+        if normalize_by == 'counts':
+            scale = norm.getSignalArray().mean()
+            self.getLogger().information('scale counts = {}'.format(int(scale)))
+        elif normalize_by == 'monitor':
+            scale = np.array(data.getExperimentInfo(0).run().getProperty('monitor_count').value)
+            scale /= norm.getExperimentInfo(0).run().getProperty('monitor_count').value[0]
+            self.getLogger().information('scale monitor = {}'.format(scale))
+        elif normalize_by == 'time':
+            scale = np.array(data.getExperimentInfo(0).run().getProperty('duration').value)
+            scale /= norm.getExperimentInfo(0).run().getProperty('duration').value[0]
+            self.getLogger().information('van time = {}'.format(scale))
+        else:
+            scale = 1
+        # exec
+        data.setSignalArray(data.getSignalArray() / scale)
+        data.setErrorSquaredArray(data.getErrorSquaredArray() / scale**2)
+        # cleanup
+        DeleteWorkspace(norm_replicated)
```

Comment:
The `DeleteWorkspace(norm_replicated)` can happen directly after `data = DivideMD(LHSWorkspace=data, RHSWorkspace=norm_replicated)`. That should reduce memory usage during this part.

Reply:
got it, moving the statement now